credit-risk/src/app.py

```diff
@@ -1,28 +1,234 @@
+        ]
+    )
+    q.page['header'] = ui.header_card(
+        box='header',
+        title='Credit Card Risk',
+        subtitle='Review customer ability to pay credit card bills',
+        icon='PaymentCard',
+        nav=[
+            ui.nav_group(
+                'Navigation', 
+                items=[
+                    ui.nav_item(name='render_customer_selector', label='Customers'),
+                ]
+            ),
+            ui.nav_group(
+                'Options', 
+                items=[
+                    ui.nav_item(name='dark_mode', label='Dark Mode'),
+                    ui.nav_item(name='light_mode', label='Light Mode'),
+                ]
+            )
+        ]
+    )
+
+
+def handle_approve_click(q: Q):
+    q.app.customer_df.loc[q.app.customer_df[ID_COLUMN] == q.client.selected_customer_id, 'Status'] = 'Approved'
+
+
+def handle_reject_click(q: Q):
+    q.app.customer_df.loc[q.app.customer_df[ID_COLUMN] == q.client.selected_customer_id, 'Status'] = 'Rejected'
+
+
+async def render_customer_page(q: Q):
+
+    del q.page['customer_table']
+
+    row_index = int(q.args.customer_table[0])
+    customer_row = q.app.customer_df.loc[row_index]
+    score = q.app.predictions_df.loc[row_index]
+    approve = bool(score < APPROVAL_THRESHOLD)
+    contribs = q.app.contributions_df.loc[row_index].drop("BiasTerm")
+
+    q.client.selected_customer_id = customer_row["ID"]
+
+    # details
+    q.page.add(
+        'customer_features', 
+        ui.form_card(
+            box='customer_features',
+            items=[
+                ui.table(
+                    name='customer_features',
+                    columns=[
+                        ui.table_column(name="attribute", label="Attribute", sortable=False, searchable=False, max_width='100'),
+                        ui.table_column(name="value", label="Value", sortable=False, searchable=False, max_width='100')
+                    ],
+                    rows=[ui.table_row(name=index, cells=[index, row]) for index, row in customer_row.map(str).iteritems()],
+                    groupable=False,
+                    resettable=False,
+                    multiple=False,
+                    height='760px'
+                )
+            ]
+        )
+    )
+
+    # summary
+    top_feature = contribs.idxmin(axis=1) if approve else contribs.idxmax(axis=1)
+    explanation_data = {
+        'will_or_will_not': 'will' if approve else 'will not',
+        'top_contributing_feature': top_feature,
+        'value_of_top_contributing_feature': str(customer_row[top_feature]),
+        'accept_or_reject': 'approve' if approve else 'reject',
+    }
+    explanation = (
+        "- This customer **{{will_or_will_not}}** most probably settle the next month credit card balance.\n"
+        "- Having a **{{top_contributing_feature}}** of **{{value_of_top_contributing_feature}}** is the top reason for that.\n"
+        "- It's a good idea to **{{accept_or_reject}}** this customer." 
```

Comment:
I would change "a good idea" to "recommended"

Reply:
Done